examples/platforms/simulation/radio.c

```diff
@@ -221,6 +222,53 @@ static uint16_t crc16_citt(uint16_t aFcs, uint8_t aByte)
     return (aFcs >> 8) ^ sFcsTable[(aFcs ^ aByte) & 0xff];
 }
 
+
+#if OPENTHREAD_SIMULATION_VIRTUAL_TIME == 1
+const char * radioStateToString(otRadioState aState)
+{
+    switch (aState)
+    {
+    case OT_RADIO_STATE_RECEIVE:
+        return "rx";
+    case OT_RADIO_STATE_TRANSMIT:
+        return "tx";
+    case OT_RADIO_STATE_DISABLED:
+        return "off";
+    case OT_RADIO_STATE_SLEEP:
+        return "sleep";
+    default:
+        fprintf(stderr, "Invalid radio status: %d\n", sState);
+        return "";
+    }
+}
+
+void reportRadioStatusToOtns(void)
+{
+    struct Event event;
+    
+    event.mDelay      = 1; // 1us for now
```

Comment:
This is following the same use of another [part of the code](https://github.com/openthread/openthread/blob/119df62cbddce9d3e10b0a56d7fbf9d53c7fa218/examples/platforms/simulation/radio.c#L843-L846):
```
void radioTransmit(struct RadioMessage *aMessage, const struct otRadioFrame *aFrame)
{
#if OPENTHREAD_SIMULATION_VIRTUAL_TIME == 0
    ssize_t            rval;
    struct sockaddr_in sockaddr;

    memset(&sockaddr, 0, sizeof(sockaddr));
    sockaddr.sin_family = AF_INET;
    inet_pton(AF_INET, OT_RADIO_GROUP, &sockaddr.sin_addr);

    sockaddr.sin_port = htons((uint16_t)(9000 + sPortOffset));
    rval =
        sendto(sTxFd, (const char *)aMessage, 1 + aFrame->mLength, 0, (struct sockaddr *)&sockaddr, sizeof(sockaddr));

    if (rval < 0)
    {
        perror("sendto(sTxFd)");
        exit(EXIT_FAILURE);
    }
#else  // OPENTHREAD_SIMULATION_VIRTUAL_TIME == 0
    struct Event event;

    event.mDelay      = 1; // 1us for now
    event.mEvent      = OT_SIM_EVENT_RADIO_RECEIVED;
    event.mDataLength = 1 + aFrame->mLength; // include channel in first byte
    memcpy(event.mData, aMessage, event.mDataLength);

    otSimSendEvent(&event);
#endif // OPENTHREAD_SIMULATION_VIRTUAL_TIME == 0
}
```
I guessed that it could be due to a delay to register the event; hence, the 1us. To avoid problems that I was not aware now, I used the same parameters.

Reply:
The 1us delay in `radioTransmit` is for simulating a radio frame transmission delay. 
We should not need the delay for radio state transition ?

Reply:
I actually prefer to set it to 0 (already did in the last commit).

Yet, I don't see how this `delay from one node to another` could be 1us in a 2.4GHz communication link (250kbps). Considering the minimal package (ACK), containing around 7-11 bytes (if we count the preamble), it would still take more than 224us to transmit it, propagate, and receive. But, this is not the point of our code here. I didn't find this delays considered in the OTNS either, but I also don't know how much the UDP layer actually delays here, I am not contesting.

Reply:
You are totally right. I actually have a PR that tries to calculate more accurate delays for frame transmission at https://github.com/openthread/openthread/pull/6476

However, the PR does not work yet. Neither do we have such implementation in OTNS. 

Again, the delay is for frame transmission and not suitable for radio state.